index.rst

```diff
@@ -53,10 +53,362 @@
+
+Kafka and Zookeeper should be listed when running
+
+.. code-block:: python
+
+  docker service ls
+
+
+End consumers
+-------------
+
+Sample consumers can be started by running either ``printStream.py``,
+which prints alert contents to the screen,
+or ``monitorStream.py``, which prints the status of the latest offset
+(number of alerts received).
+Both of these scripts are found in the ``alert_stream`` GitHub rep.
+
+To run the mini-broker prototype at full scale with 100 end-users,
+100 containers would need to be deployed, each consuming a topic
+following the naming scheme
+``Filter001``, ``Filter002``, etc... ``Filter100``.
+
+To deploy, e.g., the monitoring script using Docker on a single node,
+consuming the stream for the 10th filter, run:
+
+.. code-block:: python
+
+  docker run -it --rm \
+             --name=monitor010 \
+             --network=alert_stream_default \
+             alert_stream python bin/monitorStream.py Filter010
+
+Output is directed to the screen.
+
+To deploy the same process as a Swarm service, instead run:
+
+.. code-block:: python
+
+  docker service create \
+          --name monitor010 \
+          --network alert_stream_default \
+          --constraint node.role==worker \
+          -e PYTHONUNBUFFERED=0 \
+          alert_stream python bin/monitorStream.py Filter010
+
+
+Filters
+-------------
+
+Each individual filter is written as a class containing a function
+that operates on the contents on an alert and returns true or false.
+Filters are added by adding additional classes to ``filters.py``.
+The filtering code limits the number of passing alerts to 20
+alerts per visit.
+
+In this prototype, each filter class name should include the
+3 digit filter number, following the format of ``Filter001``,
+etc., as the filter class name is the name of the filtered
+stream from which the end-consumers read.
+
+The filtering code takes as input a range of numbers of filters
+to run at once, which consume a single instance of the unfiltered
+stream in parallel.
+For example, running
+
+.. code-block:: python
+
+    python bin/filterStream.py my-stream 1 10
+
+will deploy filters 1 through 10 in a group.
+Each group needs to read its own instance of the full stream.
+To avoid performance issues which will result in lagging
+filters, the number of groups (i.e., the number of instances
+of the full stream flowing in this system) should be kept
+to a minimum.
+See DMTN-028 :cite:`DMTN-028`.
+In this prototype, we recommend here to run five filter
+groups, each acting on a group of 20 filters
+(i.e., 1-20, 21-40, 41-60, 61-80, 81-100).
+Each group of filters can be deployed in its own
+Docker container.
+
+To deploy, e.g., the first group of filters on an
+unfiltered stream called ``full-stream`` on a single node,
+run the following:
+
+.. code-block:: python
+
+    docker run -it --rm \
+               --network=alert_stream_default \
+               alert_stream python bin/filterStream.py full-stream 1 20
+
```

Comment:
when I run this I get an error:

```docker run -it --rm --network=alert_stream_lsst_default alert_stream python bin/filterStream.py full-stream 1 20
usage: filterStream.py [-h] [--group GROUP] [--stampDir STAMPDIR]
                       [--decode | --decode-off]
                       topic
filterStream.py: error: unrecognized arguments: 1 20
```

Reply:
Sorry, this (and everything in the technote) will only work after DM-13476 is merged.  It should work if you check out that branch.